server/src/main/java/dev/slimevr/tracking/processor/skeleton/LegTweaks.java

```diff
@@ -932,6 +1148,31 @@ private void updateLockStateCounters() {
 		}
 	}
 
+	// remove the x and z components of the given quaternion
+	private Quaternion isolateYaw(Quaternion quaternion) {
+		return new Quaternion(
+			0,
+			quaternion.getY(),
+			0,
+			quaternion.getW()
+		);
+	}
```

Comment:
Is this safe? I'm not quite sure that this is fully reliable, particulatly at a rotation of 180 degrees on the yaw axis? I need someone else's input on this or if you can explain it, I would appreciate that a lot.

Reply:
I wish I could tell you but I'm not sure if it is or is not safe. If it isn't I have yet to run in to that case in testing

Reply:
we need @AxisAngles in the thread :D

Reply:
When the yaw is at 180 degrees, y will be exactly 1 or -1. So that case is completely safe.
The only time it is not safe is when the quaternion's up direction is facing exactly down. At that point, y and w will both be 0, and the resulting quaternion is null. However, this does not tend to happen in reality, so this is pretty safe.
When the bad case is encountered, there are infinitely many rotations which satisfy the "isolateYaw" constraint, and so it is reasonable and acceptable to return the identity quaternion.

Reply:
thank you for your wisdom

Reply:
Sounds good, thanks for clarifying!